compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -485,18 +488,22 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
             })
             .map(|(_, source_file)| {
                 let mut adapted = match source_file.name {
-                    // This path of this SourceFile has been modified by
-                    // path-remapping, so we use it verbatim (and avoid
-                    // cloning the whole map in the process).
-                    _ if source_file.name_was_remapped => source_file.clone(),
```

Comment:
This probably also constitutes a bug in the existing codebase since #72767 because the `source_file`'s `RealFileName` might be `Devirtualized` and thus contains the `local_path`.